lib/blockchain.py

```diff
@@ -419,89 +418,153 @@ def get_hash(self, height):
             return hash_header(self.read_header(height), height)
 
     def get_target(self, height, headers=None):
-        if NetworkConstants.TESTNET:
-            new_target = 0
-        elif height == 0:
+        if headers is None:
+            headers = {}
+
+        # Check for genesis
+        if height == 0:
             new_target = NetworkConstants.POW_LIMIT_LEGACY
+        # Check for valid checkpoint
         elif height % difficulty_adjustment_interval() == 0 and 0 <= ((height // difficulty_adjustment_interval()) - 1) < len(self.checkpoints):
             h, t = self.checkpoints[((height // difficulty_adjustment_interval()) - 1)]
             new_target = t
-        elif is_postfork(height):
-            new_target = self.get_postfork_target(height, headers)
+        # Check for prefork
+        elif not is_postfork(height):
```

Comment:
To keep consistent with the code below, better use `height < NetworkConstants.BTG_HEIGHT` instead.